providers/writing_vcr_specs.md

```diff
@@ -22,34 +22,71 @@ The next thing we have to take care of is hiding "secrets".  Since the VCR YAML
 
 VCR handles this with the `config.define_cassette_placeholder` option.  You provide VCR with a string that you want to be replaced, and then what you want it to be replaced with.  This allows for hostnames / passwords / etc... to be used when recording the cassette but the values will not be written to the resulting YAML files.
 
-ManageIQ has a pattern to help you with this, simply create a `config/secrets.defaults.yml` file:
+ManageIQ has a pattern to help you with this.  We use rails credentials.  Run the following command in the main application directory:
+
+```
+EDITOR=vi be rails credentials:edit --help
+```
+
+This will provide more information about rails credentials.
+
+A sample workflow for VCR cassettes would be to edit the test environment's credentials and use them in test.
+
+Run the prior command without the help option and with the 'test' environment:
+
+```
+EDITOR=vi be rails credentials:edit --environment test
+```
+
+You can use your preferred editor by specifying it on the command line.
+
+In the editor, set your credentials, for example:
+
 ```yaml
 ---
-test:
-  awesome_cloud_defaults: &awesome_cloud_defaults
+  awesome_cloud:
     access_key: AWESOME_CLOUD_ACCESS_KEY
     secret_key: AWESOME_CLOUD_SECRET_KEY
-  awesome_cloud:
-    <<: *awesome_cloud_defaults
 ```
 
-Then create a `config/secrets.yml` file (this file will not be committed and should be in your .gitignore):
-```yaml
----
-test:
-  awesome_cloud:
-    access_key: "YOUR_REAL_ACCESS_KEY"
-    secret_key: "YOUR_REAL_SECRET_KEY"
+After saving, this will create or update:
+* a plain text key file if you haven't already created one: config/credentials/test.key
+* an encrypted credentials file: config/credentials/test.yml.enc
+
+Add memoized methods in spec/spec_helper.rb with the defaults you want to assumed when running from cassettes to avoid leaking actual credentials:
+
+```ruby
+def credentials_awesome_cloud_host
+  @credentials_awesome_cloud_host ||= Rails.application.credentials.awesome_cloud_host || "awesome-cloud-host"
+end
+
+def credentials_awesome_cloud_user
+  @credentials_awesome_cloud_user ||= Rails.application.credentials.awesome_cloud_user || "awesome-cloud-user"
```

Comment:
Should this be `Rails.application.credentials.awesome_cloud&.access_key`

Reply:
Thanks, I'm trying to generalize them.  Does that make sense for a provider that uses user/password authentication?  If so, I can do that.

Reply:
Well two things, one I think even if there was a `user: "..."` it would be `Rails.application.credentials.awesome_cloud.user` not `Rails.application.credentials.awesome_cloud_user` and two the example here has `access_key` not `user`

I don't think I'd want to change the "writing a provider" example to have host/user/password because that doesn't match with the rest of the example code (there's an example connect method and an example api_client, etc...)

Reply:
Maybe in addition to editing this specific example to work with the new credentials interface we need a "recording_vcrs.md" guide that covers how to do this more generally?